### PR TITLE
Change log level for health_check

### LINKED ISSUE #17856: allow to set loglevel for healthcheck
### Feature request description

Currently our server logs are mostly full of healthcheck output, making them hard to navigate.

### Suggest potential solution

I would want the healthcheck service to run with `LogLevelMax=notice`, this would remove the normal output, inclusive the started/stopped messages from systemd itself.

I believe this should be the default, but there could be an option for `podman run` to override the loglevel for healtchecks.

### Have you considered any alternatives?

Filtering the log, but this is cumbersome.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cgiradkar**
Filter health_check and exec events for logging in console

Podman server logs are mostly full of healthcheck output, making them hard to navigate. Hence, made healthcheck service to run with `LogLevelMax=notice,` this would remove the normal output, inclusive the started/stopped messages from systemd itself.

Fixes #17856 

#### Does this PR introduce a user-facing change?

```release-note
Podman events' logs in systemd will output limited events in following ways:

- Podman healthcheck run would not emit `exec` and `exec_died` events

```

### CODE REVIEW

File: test/e2e/healthcheck_run_test.go
Comment (edsantiago) [inline test/e2e/healthcheck_run_test.go:254]: I'm having a hard time understanding the purpose behind this change. This does not test your PR's functionality; it also completely destroys the purpose behind the `failed to healthy` test. Perhaps there's a better way to add a test for your PR?

**Review by edsantiago — CHANGES_REQUESTED** [review]

File: test/e2e/healthcheck_run_test.go
Comment (cgiradkar) [reply test/e2e/healthcheck_run_test.go:254]: @edsantiago i agree with you, the passing criteria for the PR are somewhat complicated and this is just a preliminary push, a new test needs to be written after restoring this test (this was just a temporary "make it work").
The current use cases that needs to be checked are as follows:

- Suppress systemd logs by using `LogLevelMax=notice`
- suppress logs of the categories having events like `exec-died`

and both use cases need to be tested.

**Comment by mheon** [comment]
Code change LGTM, test needs work

File: test/e2e/healthcheck_run_test.go
Comment (Luap99) [reply test/e2e/healthcheck_run_test.go:254]: For the test you can add another podman events call like that
`podman events --filter event=exec_died --filter event=exec --since 1m`

Then you can make sure this output is empty which means your change is working an were are not getting any exec events when we run the healthcheck.

For the systemd logs I don't think it is worth to add a test for it, most tests manually run the healthcheck with `podman healtcheck run` and do not rely on the systemd timer so there is no good way to test it right now.

**Review by Luap99 — COMMENTED** [review]
code looks good

File: test/e2e/healthcheck_run_test.go
Comment (Luap99) [inline test/e2e/healthcheck_run_test.go:272]: you should add a comment and make clear that we do not expect any exec events from healthchecks, we only have these two because we run one podman exec command above

Comment (cgiradkar) [reply test/e2e/healthcheck_run_test.go:274]: done

**Comment by edsantiago** [comment]
Halfhearted suggestion:
```go
		// Test hc logs to not include exec-died events
		hcLogs := podmanTest.Podman([]string{"events", "--stream=false", "--filter", "container=hc", "--since", "1m"})
		hcLogs.WaitWithDefaultTimeout()
		Expect(hcLogs).Should(ExitCleanly())
		Expect(hcLogs.OutputToString()).ToNot(ContainSubstring("exec_died"))
```
"halfhearted" because this test actually doesn't work: I see `exec_died` events in my log. Offering it as a start.

(A better test would use the CID or a pseudorandom container name, `"hc" + random_string`, but that's left as an exercise for the reader)

**Comment by Luap99** [comment]
> ```go
> "--filter", "container=h
> ```

Because we expect one exec/exec_died event as there is an podman exec call above in the test. That is why we should test for exactly two matches as these are expected. It is just the healthcheck calls should not add more.

**Comment by Luap99** [comment]
@cgiradkar Please make sure you add more context in the commit message, the one from th ePR is fine but it should match the commit message. And also add the `Fixes #17856` to the commit message and PR description so the issue correctly closed once thi gets merged.

File: libpod/healthcheck_linux.go
Comment (edsantiago) [inline libpod/healthcheck_linux.go:31]: Is this a leftover?

File: test/e2e/healthcheck_run_test.go
Comment (edsantiago) [inline test/e2e/healthcheck_run_test.go:269]: You might want to `defer (podman stop hc)`. As a general rule it's a good idea to do your own cleanup, not leave it for teardown to have to do.

**Review by edsantiago — COMMENTED** [review]
Much better. One question, one suggestion, neither blocking.

File: libpod/healthcheck_linux.go
Comment (cgiradkar) [reply libpod/healthcheck_linux.go:31]: No, it was introduced as planned. (assuming by leftover, you meant code that need not be there)

File: test/e2e/healthcheck_run_test.go
Comment (cgiradkar) [reply test/e2e/healthcheck_run_test.go:270]: done, thanks!

File: libpod/healthcheck_linux.go
Comment (edsantiago) [reply libpod/healthcheck_linux.go:31]: Thank you. If it's deliberate, would it not be better to define it in the declaration? I don't know Go best practices, but `declare empty, then append` looks jarring to my eye. It costs me energy, trying to figure out why it was done that way, and will presumably be a burden to future maintainers. But perhaps Go does things differently?

File: test/e2e/healthcheck_run_test.go
Comment (edsantiago) [reply test/e2e/healthcheck_run_test.go:270]: I'm sorry, I used unwarranted shortcuts in my hasty comment. Plain `stop` is a very bad thing to do on a container running `sleep`: it will waste ten seconds (and those seconds add up over the course of thousands of tests). Also, there's no `Wait` or status check.

I've looked in the code base for an example to emulate, and find none, so it looks like `defer` is not the right approach. And other tests in this module don't bother cleaning up, so I retract my request with apologies. Just remove the `defer`, let teardown take care of cleanup.

File: libpod/healthcheck_linux.go
Comment (cgiradkar) [reply libpod/healthcheck_linux.go]: removed

Comment (cgiradkar) [reply libpod/healthcheck_linux.go]: adjusted in code, thanks!

File: test/e2e/healthcheck_run_test.go
Comment (cgiradkar) [reply test/e2e/healthcheck_run_test.go:269]: removed the line, thanks

File: libpod/container_exec.go
Comment (Luap99) [inline libpod/container_exec.go:1117]: these must be set true, no? Still PR still faces Doesn't this PR still introduce https://github.com/containers/podman/issues/20188 so this must be fixed before the merge to not cause regressions.

Comment (cgiradkar) [reply libpod/container_exec.go]: passed true value as discussed.

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**
